engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/requestresponse/RequestResponseRunnableScenarioInterpreter.scala

```diff
@@ -29,28 +26,16 @@ class RequestResponseRunnableScenarioInterpreter(jobData: JobData,
 
   import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter._
 
-  private val akkaHttpCloseTimeout = 10 seconds
-
   private var closed: Boolean = false
 
   @volatile private var interpreter: RequestResponseScenarioInterpreter[Future] = _
 
-  @volatile private var server: ServerBinding = _
-
   override def run(): Future[Unit] = {
-    interpreter = RequestResponseInterpreter[Future](parsedResolvedScenario, jobData.processVersion, contextPreparer, modelData, Nil, ProductionServiceInvocationCollector, ComponentUseCase.EngineRuntime)
+    val i = RequestResponseInterpreter[Future](parsedResolvedScenario, jobData.processVersion, contextPreparer, modelData, Nil, ProductionServiceInvocationCollector, ComponentUseCase.EngineRuntime)
       .valueOr(errors => throw new IllegalArgumentException(s"Failed to compile: $errors"))
-    interpreter.open()
-    val path = ScenarioRoute.pathForScenario(jobData.metaData).getOrElse(parsedResolvedScenario.id) // TODO: path should be required
-    val route = new ScenarioRoute(Map(path -> new RequestResponseAkkaHttpHandler(interpreter)), requestResponseConfig.definitionMetadata)
-    implicit val materializer: Materializer = Materializer(actorSystem)
-    logger.info(s"Binding scenario route into ${requestResponseConfig.interface}:${requestResponseConfig.port}")
-    Http().newServerAt(
-      interface = requestResponseConfig.interface,
-      port = requestResponseConfig.port
-    ).bind(route.route).map { binding =>
-      logger.info("Scenario route bound")
-      server = binding
+    i.open()
+    interpreter = i
```

Comment:
I wonder if it would be better to move this all interpreter initialization to class body and make routes just a val (here - not in interface). Thanks to that, this class will be very easy, there won't be any constraint between order of execution of run and routes, status would just return TaskStatus.Running. WDYT?

Reply:
I tought about it, but it `ScenarioInterpreterFactory.createInterpreter` compiles scenario, runs validation etc.. 

But let's try.

Reply:
:+1: It almost ok. Originally I was thinking about moving interpreter.open() to class body. After that it won't be any problem with returning TaskStatus.Running in status. I see two options:
1. We move interpreter.open() to class body
2. We add isOpened in `Lifecycle` or in `RequestResponseRunnableScenarioInterpreter` and we have `if (interpreter.isOpened) TaskStatus.Running else TaskStatus.DuringDeploy`
WDYT?

Reply:
Right.
Lets go with option 1.